src/plugins/GUITestBase/src/tests/regression_scenarios/GTTestsRegressionScenarios_6001_7000.cpp

```diff
@@ -6822,6 +6823,35 @@ GUI_TEST_CLASS_DEFINITION(test_6995) {
     CHECK_SET_ERR(visibleRange.contains(6151), "Complement read is not centered: " + visibleRange.toString());
 }
 
+GUI_TEST_CLASS_DEFINITION(test_6999) {
+    // 1. Create read_only_dir
+    const QString projectPath = QFileInfo(sandBoxDir + "read_only_dir/project.uprj").absoluteFilePath();
+
+    QDir().mkpath(sandBoxDir + "read_only_dir");
+    GTFile::setReadOnly(os, sandBoxDir + "read_only_dir");
+
+    // 2. Open any file
+    GTFileDialog::openFile(os, dataDir + "samples/FASTA/human_T1.fa");
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+
+    GTUtilsDialog::waitForDialog(os, new MessageBoxDialogFiller(os, QMessageBox::StandardButton::Ok));
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+    GTUtilsDialog::waitForDialog(os, new SaveProjectAsDialogFiller(os, "New Project", projectPath));
+    GTUtilsTaskTreeView::waitTaskFinished(os);
+
+    // 3. Select "File->Save project as..."
+    // 4. Save project to file "read_only_dir/project.proj"
+    // 5. Click "Save"
+    //    Expected state: Message Box with text "Folder is read-only" appears
+    GTMenu::clickMainMenuItem(os, QStringList() << "File"
+                                                << "Save project as...");
+
+    QWidget *dialog = QApplication::activeModalWidget();
```

Comment:
Try using GTWidget::getActiveModalWidget. It will wait until the widget appears and will never return nullptr.